File: Rentify.Services/Database/Reservation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Rentify.Services.Database
{
    public class Reservation
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey(nameof(UserId))]
        public int UserId { get; set; }
        public User? User { get; set; }

        [ForeignKey(nameof(PropertyId))]
        public int PropertyId { get; set; }
        public Property? Property { get; set; }
        public bool IsMonthly { get; set; }
        public bool? IsApproved { get; set; }
        public DateTime CreatedAt {get; set;}
        public DateTime? StartDateOfRenting {get; set;}
        public DateTime? EndDateOfRenting {get; set;}

    }
}

File: UI/rentify_mobile/lib/models/reservation.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:rentify_mobile/models/property.dart';
import 'package:rentify_mobile/models/user.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  final int id;
  final int userId;
  final User? user;
  final int propertyId;
  final Property? property;
  final bool isMonthly;
  bool? isApproved;
  DateTime? createdAt;
  DateTime? startDateOfRenting;
  DateTime? endDateOfRenting;

  Reservation({
    required this.id,
    required this.userId,
    required this.user,
    required this.propertyId,
    required this.property,
    required this.isMonthly,
    this.isApproved,
    this.createdAt,
    this.startDateOfRenting,
    this.endDateOfRenting,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
